libs/cardano-ledger-core/testlib/Test/Cardano/Ledger/Imp/Common.hs

```diff
@@ -212,6 +214,16 @@ shouldEndWith x y = liftIO $ H.shouldEndWith x y
 shouldContain :: (HasCallStack, Show a, Eq a, MonadIO m) => [a] -> [a] -> m ()
 shouldContain x y = liftIO $ H.shouldContain x y
 
+shouldContainExpr :: (HasCallStack, ToExpr a, Eq a, MonadIO m) => [a] -> [a] -> m ()
+shouldContainExpr x y
+  | y `isInfixOf` x = pure ()
+  | otherwise =
+      assertFailure $
+        "Left does not contain right:\nLeft:\n"
```

Comment:
I think using `Left` and `Right` here is a little misleading. In the rest of these functions they are used with `Either`, but here it's nothing to do with that, no?